bench/irmin-pack/parse_log.ml

```diff
@@ -0,0 +1,129 @@
+    match Sys.command cmd with
+    | 0 -> ()
+    | n ->
+        Fmt.failwith
+          "Failed to set up the test environment: command `%s' exited with \
+           non-zero exit code %d"
+          cmd n)
+  else Fmt.failwith "File %s not found" file
+
+let operation_list = ref []
+let exact_string op = regexp ("\\[" ^ op ^ "\\]")
+let comma = regexp ","
+let space = regexp " "
+
+let string_after_regexp op s =
+  let pos = search_forward (exact_string op) s 0 in
+  let x = matched_string s |> String.length in
+  string_after s (pos + x + 1)
+
+let extract_op s op f =
+  try
+    string_after_regexp op s |> split space |> function
+    | key :: tl ->
+        let keys = split comma key in
+        let v = String.concat " " tl in
+        operation_list := f keys v :: !operation_list;
+        true
+    | _ -> Fmt.failwith "Unexpected arguments for %s : %s" op s
+  with Not_found -> false
+
+let set s = extract_op s "raw set" (fun k v -> Add (k, v))
+
+let bool = function
+  | "true" -> true
+  | "false" -> false
+  | v -> Fmt.failwith "Unexpected arguments %s" v
+
+let find s = extract_op s "find" (fun k v -> Find (k, bool v))
+let mem s = extract_op s "mem" (fun k v -> Mem (k, bool v))
+let mem_tree s = extract_op s "mem_tree" (fun k v -> Mem_tree (k, bool v))
+
+let extract_op s op f =
+  try
+    let op = string_after_regexp op s |> split comma |> f in
+    operation_list := op :: !operation_list;
+    true
+  with Not_found -> false
+
+let remove s =
+  extract_op s "remove" (fun x -> Remove x)
+  || extract_op s "raw del" (fun x -> Remove x)
+
+let checkout s =
+  try
+    let _ = string_after_regexp "checkout" s in
+    operation_list := Checkout :: !operation_list;
+    true
+  with Not_found -> false
+
+let substring from to_ s =
+  try
+    let from = exact_string from in
+    let to_ = exact_string to_ in
+    let pos = search_forward from s 0 in
+    let x = matched_string s |> String.length in
+    let s = string_after s (pos + x + 1) in
+    let pos = search_forward to_ s 0 in
+    string_before s (pos - 1)
+  with Not_found -> Fmt.failwith "substring %s %s %s" from to_ s
+
+let commit s =
+  let _hash = substring "commit" "date" s in
+  let date = substring "date" "message" s |> Int64.of_string in
+  let message = substring "message" "parents" s in
+  let _parents = string_after_regexp "parents" s |> split comma in
+  let op = Commit (date, message) in
+  operation_list := op :: !operation_list
+
+let copy s =
+  if string_match (exact_string "copy from") s 0 then (
+    let from = substring "copy from" "to" s |> split comma in
+    let to_ = string_after_regexp "to" s |> split comma in
+    operation_list := Copy (from, to_) :: !operation_list;
+    true)
+  else false
+
+let extract_operation s =
+  if
+    not
+      (set s
+      || remove s
+      || find s
+      || mem s
+      || mem_tree s
+      || checkout s
+      || copy s)
+  then Fmt.failwith "Unexpected operation %s" s
+
+let read_commit ic commits i =
```

Comment:
You should have picked something like json for your log, it would have been less painful to parse :)

Reply:
Indeed :) switched to a json trace.